crates/ruff/src/rules/flake8_todo/rules.rs

```diff
@@ -0,0 +1,525 @@
+            // TDO-006
+            let mut invalid_capitalization = Diagnostic::new(
+                InvalidCapitalizationInTodo {
+                    tag: tag.content.to_string(),
+                },
+                tag.range,
+            );
+
+            if autofix.into() && settings.rules.should_fix(Rule::InvalidCapitalizationInTodo) {
+                invalid_capitalization.set_fix(Fix::unspecified(Edit::replacement(
+                    "TODO".to_string(),
+                    tag.range.start(),
+                    tag.range.end(),
+                )));
+            }
+
+            diagnostics.push(invalid_capitalization);
+        }
+    }
+}
+
+/// Checks for "static" errors in the comment - missing colon, missing author, etc. This function
+/// modifies `diagnostics` in-place.
+fn check_for_static_errors(
+    comment: &str,
+    comment_range: TextRange,
+    tag: &Tag,
+    diagnostics: &mut Vec<Diagnostic>,
+) {
+    // Relative offset of the current character from the start of the comment.
+    let mut relative_offset: usize = usize::try_from(tag.range.end() - comment_range.start())
+        .ok()
+        .unwrap();
+    let mut comment_chars = comment.chars().skip(relative_offset).peekable();
+    // Absolute offset of the comment's author block from the start of the file.
+    let mut author_range: Option<TextRange> = None;
+    // Absolute offset of the comment's colon from the start of the file.
+    let mut colon_offset: Option<TextSize> = None;
+
+    // An "author block" must be contained in parentheses, like "(ruff)". To check if it exists,
+    // we can check the first non-whitespace character after the tag. If that first character is a
+    // left parenthesis, we can say that we have an author's block.
+    for char in comment_chars.by_ref() {
+        relative_offset += 1;
+        if char.is_whitespace() {
+            continue;
+        }
+
+        // We can guarantee that there's no author if the colon directly follows the TODO tag.
+        if char == ':' {
+            colon_offset =
+                Some(comment_range.start() + TextSize::try_from(relative_offset).ok().unwrap());
+            break;
+        }
+
+        if char == '(' {
+            author_range = Some(TextRange::at(
+                comment_range.start() + TextSize::try_from(relative_offset).ok().unwrap(),
+                TextSize::new(1),
+            ));
+        }
+
+        break;
+    }
+
+    if let Some(range) = author_range {
+        let mut author_block_length = 0usize;
+        for char in comment_chars.by_ref() {
+            relative_offset += 1;
+            author_block_length += 1;
+
+            if char == ')' {
+                break;
+            }
+        }
+
+        author_range = Some(range.add_end(TextSize::try_from(author_block_length).ok().unwrap()));
+    } else {
+        diagnostics.push(Diagnostic::new(
+            MissingAuthorInTodo,
+            TextRange::at(tag.range.end(), TextSize::new(1)),
```

Comment:
Using `TextSize::new(1)` as length here is unsafe in case the first character after the TODO is a non-ascii character, because it then only includes the first byte of the multi-byte character. 

What you can do is either compute the size of the next character `text_after_range_end.chars().next().map(TextLen::text_len).unwrap_or_default()` or we change the range to highlight the tag (I know, that's the opposite of what I said last time, I'm sorry). Highlighting the tag might be easier. 

This is also true for other places where you use `TextSize::new(1)` as the range's length

Reply:
Ah I see what you mean - okay, yeah. I'll just stick to pointing to the `tag` for now, that seems much easier